Tests/ApolloPaginationTests/SubscribeTests.swift

```diff
@@ -0,0 +1,136 @@
+
+    server = MockGraphQLServer()
+    let networkTransport = MockNetworkTransport(server: server, store: store)
+
+    client = ApolloClient(networkTransport: networkTransport, store: store)
+    MockSchemaMetadata.stub_cacheKeyInfoForType_Object = IDCacheKeyProvider.resolver
+  }
+
+  override func tearDownWithError() throws {
+    cache = nil
+    server = nil
+    client = nil
+
+    try super.tearDownWithError()
+  }
+
+  private typealias Query = MockQuery<Mocks.Hero.FriendsQuery>
+
+  func test_multipleSubscribers() throws {
+    let query = Query()
+    query.__variables = ["id": "2001", "first": 2, "after": GraphQLNullable<String>.null]
+    let pager = GraphQLQueryPager.makeForwardCursorQueryPager(client: client) { page in
+      let query: Query = MockQuery()
+      let after: GraphQLNullable<String>
+      if let endCursor = page?.endCursor {
+        after = .some(endCursor)
+      } else {
+        after = .none
+      }
+      query.__variables = [
+        "id": "2001",
+        "first": 2,
+        "after": after
+      ]
+      return query
+    } extractPageInfo: { data in
+      CursorBasedPagination.ForwardPagination(
+        hasNext: data.hero.friendsConnection.pageInfo.hasNextPage,
+        endCursor: data.hero.friendsConnection.pageInfo.endCursor
+      )
+    }
+
+    addTeardownBlock { pager.cancel() }
+
+    let initialFetchExpectation = expectation(description: "Results")
+    initialFetchExpectation.assertForOverFulfill = false
+
+    var results: [Result<GraphQLQueryPager<Query, Query>.Output, Error>] = []
+    var otherResults: [Result<GraphQLQueryPager<Query, Query>.Output, Error>] = []
+    pager.subscribe { result in
+      results.append(result)
+      initialFetchExpectation.fulfill()
+    }
+
+    pager.subscribe { result in
+      otherResults.append(result)
+    }
+
+    try runActivity("Initial fetch from server") { _ in
+      let serverExpectation = server.expect(Query.self) { _ in
+        let pageInfo: [AnyHashable: AnyHashable] = [
+          "__typename": "PageInfo",
+          "endCursor": "Y3Vyc29yMg==",
+          "hasNextPage": true
+        ]
+        let friends: [[String: AnyHashable]] = [
+          [
+            "__typename": "Human",
+            "name": "Luke Skywalker",
+            "id": "1000",
+          ],
+          [
+            "__typename": "Human",
+            "name": "Han Solo",
+            "id": "1002",
+          ]
+        ]
+        let friendsConnection: [String: AnyHashable] = [
+          "__typename": "FriendsConnection",
+          "totalCount": 3,
+          "friends": friends,
+          "pageInfo": pageInfo
+        ]
+
+        let hero: [String: AnyHashable] = [
+          "__typename": "Droid",
+          "id": "2001",
+          "name": "R2-D2",
+          "friendsConnection": friendsConnection
+        ]
+
+        let data: [String: AnyHashable] = [
+          "hero": hero
+        ]
+
+        return [
+          "data": data
+        ]
```

Comment:
Would love to know if y'all have a better solution for this -- I noticed that intermingling `string` and `int` types leads to Swift being unhappy for literals like this, which had me declare chunks as a workaround -- any thoughts?